### ISSUE TITLE
Nov 3-4: Build a web app with ASP.NET Core

### ISSUE THREAD

**Issue by mebaumb** [microsoft/ReactorPlanning#114]
# Event Information
| Category | Details |
|-----------|---------|
| Reactor Topic |  Coding, Languages, and Frameworks |
| Monthly Topic (if applicable) | .NET November |
| Workshop Topic |  ASP.NET |
| Workshop Duration | 60 min | 
| Title | Build a web app with ASP.NET Core  |
| Description | Join this workshop where we'll create a web API project with ASP.NET Core. We'll also create an in-memory database for persisting products, and test the web API action methods. It will help if you have some experience with writing C# and are familiar with REST API methods like GET, POST, PUT, and DELETE. During this workshop, we'll be going through the Learn module: https://aka.ms/ASPWebAPI-2. To follow along just click the link and sign into the Learn platform with your Microsoft account. |
| Content | https://aka.ms/ASPWebAPI-8 |

# Workshop(s) details
| Date |  Local Start Time |  Location |  Speaker | SharePoint ID | Platform |
|------|-------------------|-----------|----------|---------------|----------|
|11/3 | 6:30 pm | Bangalore | Devendra Rawat | 11865 |----------|
|11/3 | 4:00 pm PT | US, Canada, Sydney | Cody Thaller (Devon Newsom) | 11866 |----------|
| 11/4 | 7:30 pm | Sao Paulo | Rafael Dos Santos | 11867 |----------|
| 11/4 | 7:30 pm | Shanghai | Warren Wong | 11868 |----------|
| 11/4 | 12pm GMT | London, Stockholm, Abu Dhabi, Tel Aviv | Allen Sanders | 11869 |----------|

# Event Checklist

## 10 Weeks Prior
- [x] Reactor Content Team (RCT) opens a new issue for an upcoming local workshop filling in the Event Information and Workshop details tables. RCT member who owns content (content owner) this week will be the assignee.
- [x] RCT and content owner discuss details of the event in the comments of this issue. RCT adds a _confirmed-event_ label and _local-digital-event_ tag to the issue and adds event to the __Local Digital Events__ project. Issue will appear in the __Upcoming Streams__ column
- [ ] RCT adds a _ready-for-marketing-edit_ label so the RCT can review title and description

## 8 Weeks Prior
- [ ] RCT does a review of title and description and removes the _ready-for-marketing-edit_ label 
- [ ] RCT moves the issue to the __Ready for SharePoint__ column
- [ ] RCT informs the Reactor Marketing Team of the new event information
- [ ] RCT confirms the speaker
- [ ] RCT provides speaker with assets (e.g. slides, overlays) and the survey data for the event
- [ ] RCT moves the issue to the __Content Preparation In Progress__ column

## 5 weeks Prior
- [ ] Content owner opens a PR with the new content for the workshop on the [customer-facing Reactor repository](https://github.com/microsoft/Reactors/tree/main/online-event-resources) and adds the _content-development_ and _local-digital-event_ tag to the PR. 
- [ ] RCT reviews, approves, and merges the changes
- [ ] RCT shares content with the vendor/instructor
- [ ] RCT provides vendor/instructor with assets (e.g. slides, overlays) and the survey data for the event
- [ ] RCT ensures that instructors have received an invite to the workshops and that PM team works with instructors to get setup in the platform (e.g. Teams, Teams Live Event)

## 1 Week Prior
- [ ] Connect vendor/instructor with the PM(s) for the locations where this event has been marketed
- [ ] Check-in with vendor/instructor that they are prepared for the workshop
- [ ] RCT moves the issue to the __Ready to Stream__ column

## During and After Stream
- [ ] Instructor (and optional moderator) run the stream
- [ ] _(Optional)_RCT submits recording to be uploaded to the Reactor YouTube and to be made available on LearnTV
- [ ] RCT moves issue to the __Stream Complete__ column

**Comment by mebaumb** [comment]
@meaghanlewis is this event ready for publish? Are we still waiting on times to be confirmed?

**Comment by meaghanlewis** [comment]
@mebaumb yes this is ready to publish!

**Comment by mebaumb** [comment]
Events are published in Dynamics. Waiting on Speaker details.

**Comment by meaghanlewis** [comment]
Okay @mebaumb, here's another pass at the description.

Join this workshop where we'll create a cross-platform REST service to manage inventory for an online dog toy retailer.  We'll be using the ASP.NET Core web API with .NET Core CLI and C#. And then we'll setup an in-memory database to store product data. To follow along, all you'll need is a Microsoft account.

Prerequisites:
Experience with witing C# at the beginner level
Familiarity with REST API methods like GET, POST, PUT, and DELETE

**Comment by mebaumb** [comment]
Speakers are updated and published.

**Comment by meaghanlewis** [comment]
Here's an overview of what this workshop will cover and suggested agenda to follow with links and resources.

## Content to teach
This workshop will take Reactor community members through the Mirosoft Learn module: [Create a web API with ASP.NET Core](https://aka.ms/ASPWebAPI-8)

**SHARE THIS LINK WITH REACTOR ATTENDEES: https://aka.ms/ASPWebAPI-8**

**Encourage attendees to go through the modules with you and complete it. Also encourage them to rate the module at the end.**

## Agenda
1. [Intro](https://docs.microsoft.com/en-us/learn/modules/build-web-api-net-core/1-introduction) 3 min
   - Welcome community members
   - Provide overview of content to be covered
   - Explain how they can follow along during the workshop
   - Encourage community members to complete and rate the module
2. [Create a web API project](https://docs.microsoft.com/en-us/learn/modules/build-web-api-net-core/2-create-web-api) 10 min
    - Use the Azure Cloud Shell to setup the environment and scaffold the web API project
    - Confirm everyone can get their Azure Cloud Shell sandbox working successfully
3. [Add a data store](https://docs.microsoft.com/en-us/learn/modules/build-web-api-net-core/3-add-data-store) 10 min
    - Explain what the data store will do
    - Use the Cloud Shell to add a data store
4. [Add a controller](https://docs.microsoft.com/en-us/learn/modules/build-web-api-net-core/4-add-controller) 5 min
    - Explain what the controller will do
    - Follow the steps to create a controller
5. [Implement CRUD operations](https://docs.microsoft.com/en-us/learn/modules/build-web-api-net-core/5-implement-crud) 10 min
    - Explain the different CRUD operations
    - Add CRUD operations to the controller
6. [Test web API actions](https://docs.microsoft.com/en-us/learn/modules/build-web-api-net-core/6-test-web-api) 10 min
    - Talk about the importance of testing
    - Send POST requests to the web API
7. **Next steps** 5 min
    - Encourage attendees to complete the knowledge check and summary
4. **Closing** 5 min
    - Q&A
    - **Ask attendees to take the survey: https://aka.ms/Reactor/Survey**
      - Note: The event code is listed in this issue description and/or your Reactor contact can share that with the community members during the workshop
    - Share Resources
      - Encourage them to visit: https://docs.microsoft.com/learn/browse/?terms=ASP.NET for more ASP.NET content on Microsoft Learn
    - Check if the Reactor contact has other information to share

**Comment by mebaumb** [comment]
Added the learn link to the reg page on dynamics/meetup (10/23)

**Comment by mebaumb** [comment]
Cancelling  US on Nov 3. Asked another location to record the session to send to attendees.

**Comment by mebaumb** [comment]
Workshops have been run, and numbers posted in tracking document.
